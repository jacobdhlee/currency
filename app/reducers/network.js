import { CHANGE_NETWORK_STATUS } from '../actions/network';

const INITIAL_STATE = {
  connected: false,
  hasCheckedStatus: false,
};

const isConnected = (status) => {
  if (status.toLowerCase() === 'none') {
    return false;
  }
  return true;
};

const network = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case CHANGE_NETWORK_STATUS:
      return {
        ...state,
        hasCheckedStatus: true,
        connected: isConnected(action.status),
      };
    default:
      return state;
  }
};

export default network;
