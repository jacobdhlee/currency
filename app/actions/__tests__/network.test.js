import { changeNetworkStatus } from '../network';

it('create CHANGE_NETWORK_STATUS snapshot', () => {
  expect(changeNetworkStatus).toMatchSnapshot();
});
