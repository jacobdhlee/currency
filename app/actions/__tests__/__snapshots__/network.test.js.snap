// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create CHANGE_NETWORK_STATUS snapshot 1`] = `[Function]`;
